pkg/kubelet/kubelet.go

```diff
@@ -796,8 +801,21 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 
 	klet.admitHandlers.AddPodAdmitHandler(klet.containerManager.GetAllocateResourcesPodAdmitHandler())
 
-	criticalPodAdmissionHandler := preemption.NewCriticalPodAdmissionHandler(klet.GetActivePods, killPodNow(klet.podWorkers, kubeDeps.Recorder), kubeDeps.Recorder)
-	klet.admitHandlers.AddPodAdmitHandler(lifecycle.NewPredicateAdmitHandler(klet.getNodeAnyWay, criticalPodAdmissionHandler, klet.containerManager.UpdatePluginResources))
+	// Any admission handler that depends on GetNode() (or getNodeAway()) must have its addition delayed
+	// after the node sync with the apiserver. This is done to prevent caching of Node objects.
+	go func() {
```

Comment:
thinking out loud, we could extend `PodAdmitAttributes` to include `Pod` , `OtherPods` , and `Node` gett-er function in future.